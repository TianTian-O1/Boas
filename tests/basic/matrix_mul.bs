def matrix_mul(a: list[list[float]], b: list[list[float]]) -> list[list[float]]:
    if len(a[0]) != len(b):
        print("矩阵维度不匹配")
        return []
    
    rows = len(a)
    cols = len(b[0])
    n = len(b)
    
    # 初始化结果矩阵
    result = [[0.0] * cols for _ in range(rows)]
    
    # CPU版本的矩阵乘法
    for i in range(rows):
        for j in range(cols):
            for k in range(n):
                result[i][j] += a[i][k] * b[k][j]
    
    return result

# 测试用例
a = [[1.0, 2.0], [3.0, 4.0]]
b = [[5.0, 6.0], [7.0, 8.0]]

result = matrix_mul(a, b)

# 打印结果
for row in result:
    for val in row:
        print(val)