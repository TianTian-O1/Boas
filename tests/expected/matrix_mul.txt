19.0
22.0
43.0
50.0